src/docs/arch-locking-hierarchy.dox

```diff
@@ -0,0 +1,105 @@
+/*! @arch_page arch-locking-hierarchy Locks hierarchy
+
+This page describes the current nested locks hierarchy of WiredTiger.
```

Comment:
Minor, both here and line 1 I think "Locking hierarchy" flows better than "Locks hierarchy", and it matches the name of the file.

Reply:
Fixed